exp/api/v1beta1/azuremachinepool_webhook_test.go

```diff
@@ -234,6 +234,30 @@ func TestAzureMachinePool_ValidateCreate(t *testing.T) {
 			ownerNotFound: true,
 			wantErr:       true,
 		},
+		{
```

Comment:
Nice tests! Can we add another one that verifies we get an error if there're more than 4 mixes instances set?